code.gs

```diff
@@ -2026,10 +2043,13 @@ function getEventsOnDate (eventDate, calendarId) {
     log.add('The calendar with ID "' + calendarId + '" is not accessible: check your calendarId value!', Priority.FATAL_ERROR);
   }
 
+  eventDate = dateWithTimezone(year, month, day, 0, 0, 0, eventCalendar.timeZone);
+
   // Query the events calendar for events on the specified date.
   try {
-    startDate = Utilities.formatDate(eventDate, eventCalendar.timeZone, 'yyyy-MM-dd\'T\'HH:mm:ss\'Z\'');
-    endDate = Utilities.formatDate(new Date(eventDate.getTime() + 1 * 60 * 60 * 1000), eventCalendar.timeZone, 'yyyy-MM-dd\'T\'HH:mm:ss\'Z\'');
+    // Look for events from 00:00:00 to 00:01:00 of the specified day.
+    startDate = Utilities.formatDate(eventDate, eventCalendar.timeZone, 'yyyy-MM-dd\'T\'HH:mm:ssXXX');
+    endDate = Utilities.formatDate(new Date(eventDate.getTime() + 60 * 1000), eventCalendar.timeZone, 'yyyy-MM-dd\'T\'HH:mm:ssXXX');
```

Comment:
nitpick: Now that this is mathematically simplified I don't think `... + 60 * 1000` gains any clarity over `... + 60000`, while losing efficiency due to the extra op. If you do prefer it though, and prefer it enough to deem the performance-hit worthwhile, then ignore this comment.